pkg/api/db.go

```diff
@@ -66,50 +66,34 @@ func CreateVolumeDBEntry(ctx *c.Context, in *model.VolumeSpec) (*model.VolumeSpe
 	if in.CreatedAt == "" {
 		in.CreatedAt = time.Now().Format(constants.TimeFormat)
 	}
-	vol := &model.VolumeSpec{
-		BaseModel: &model.BaseModel{
-			Id:        in.Id,
-			CreatedAt: in.CreatedAt,
-		},
-		UserId:            ctx.UserId,
-		Name:              in.Name,
-		Description:       in.Description,
-		ProfileId:         in.ProfileId,
-		Size:              in.Size,
-		AvailabilityZone:  in.AvailabilityZone,
-		Status:            model.VolumeCreating,
-		SnapshotId:        in.SnapshotId,
-		SnapshotFromCloud: in.SnapshotFromCloud,
-	}
-	result, err := db.C.CreateVolume(ctx, vol)
-	if err != nil {
-		log.Error("When add volume to db:", err)
-		return nil, err
-	}
 
-	return result, nil
+	in.Status = model.VolumeCreating
+	// Store the volume data into database.
+	return db.C.CreateVolume(ctx, in)
 }
 
-func ExtendVolumeDBEntry(ctx *c.Context, volID string) (*model.VolumeSpec, error) {
+func ExtendVolumeDBEntry(ctx *c.Context, volID string, newSize int64) (*model.VolumeSpec, error) {
 	volume, err := db.C.GetVolume(ctx, volID)
 	if err != nil {
 		log.Error("Get volume failed in extend volume method: ", err)
 		return nil, err
 	}
 
 	if volume.Status != model.VolumeAvailable {
-		errMsg := "The status of the volume to be extended must be available"
+		errMsg := "The status of the volume to be extended must be available!"
+		log.Error(errMsg)
+		return nil, errors.New(errMsg)
+	}
+	if newSize <= volume.Size {
+		errMsg := fmt.Sprintf("New size for extend must be greater than current size."+
+			"(current: %d GB, extended: %d GB).", volume.Size, newSize)
```

Comment:
Error message should start with lower case letter and end without a punctuation.